pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3235,6 +3235,35 @@ public Map<String, List<String>> getServerToSegmentsMap(String tableNameWithType
     return serverToSegmentsMap;
   }
 
+  /**
+   * Get the servers to segments map for which servers are ONLINE in external view for those segments in IDEAL STATE
+   */
+  public Map<String, List<String>> getOnlineServerToSegmentsMap(String tableNameWithType) {
```

Comment:
We want to make the method name more specific to reflect this one checks EV instead of IS. We should also exclude replaced segments